hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/TemplatePathsTraverser.java

```diff
@@ -389,228 +413,89 @@ protected int accessedNodes() {
         }
     }
 
-    private class SingleTraverser extends Traverser {
-
-        private MultivaluedMap<Id, Node> sources = newMultivalueMap();
-        private MultivaluedMap<Id, Node> targets = newMultivalueMap();
-        private MultivaluedMap<Id, Node> sourcesAll = newMultivalueMap();
-        private MultivaluedMap<Id, Node> targetsAll = newMultivalueMap();
-
-        private Set<Path> paths;
+    private class ConcurrentTraverser extends Traverser {
 
-        public SingleTraverser(Collection<Id> sources, Collection<Id> targets,
-                               List<RepeatEdgeStep> steps, boolean withRing,
-                               long capacity, long limit) {
-            super(steps, capacity, limit, withRing);
-            for (Id id : sources) {
-                this.sources.add(id, new Node(id));
-            }
-            for (Id id : targets) {
-                this.targets.add(id, new Node(id));
-            }
-            this.paths = new PathSet();
+        public ConcurrentTraverser(Collection<Id> sources,
+                                   Collection<Id> targets,
```

Comment:
move to previous line

Reply:
move to previous line will exceed 80 char